pkg/pipe/pipe_factory_test.go

```diff
@@ -476,12 +476,12 @@ func Test_NumExecs(t *testing.T) {
 	ctx := context.Background()
 	cmd := exec.Command("php", "../../tests/client.php", "echo", "pipes")
 
-	w, _ := NewPipeFactory().SpawnWorkerWithContext(ctx, cmd)
+	w, _ := NewPipeFactory().SpawnWorkerWithTimeout(ctx, cmd)
```

Comment:
Does it make sense to pass time.Duration instead of CTX since it named like that?

Reply:
Yes, this is the first step to reduce context usage. I have removed context from the Close, Kill, Destroy and etc. functions.